### PR TITLE
build: Shifted CI from Travis to Github Actions

### PR DESCRIPTION
**Pull request by Jawayria**
This is a part of Django 3.2 upgrade process as mentioned in this doc [Django 3.2 Upgrade: Steps to Upgrade a Repository](https://openedx.atlassian.net/wiki/spaces/AC/pages/1704591765/Django%2B3.2%2BUpgrade%2BSteps%2Bto%2BUpgrade%2Ba%2BRepository)

JIRA: https://openedx.atlassian.net/browse/BOM-2732

### CODE REVIEW

File: requirements/pip.in [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Core dependencies for installing other packages
+
+pip==20.0.2
```
Comment (UsamaSadiq) [inline requirements/pip.in:3]: Why don't we use the latest pip version instead?

**Review by UsamaSadiq — APPROVED** [review]

File: requirements/pip.in (hunk 1 above)
Comment (Jawayria) [reply requirements/pip.in:3]: This version was being explicitly installed in travis config file previously. Using latest version raises errors.

**Comment by Jawayria** [comment]
> BTW: this should not be a "feat:" commit: it doesn't change the feature behavior of the code. This is a "build:" commit because it's about the infrastructure for building or testing the code.

Okay I'll update the commit message

**Review by mraarif — APPROVED** [review]
should do a rebase and fix conflicts

File: .github/workflows/pypi-publish.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+name: Publish package to PyPi
+
+on:
+  release:
+    types: [published]
```
Comment (nedbat) [inline .github/workflows/pypi-publish.yml:5]: This should be based on a tag, not a GitHub release.

**Review by nedbat — CHANGES_REQUESTED** [review]

**Review by nedbat — APPROVED** [review]

**Pull request merged**